### PR TITLE
Host Guide Edits: Fix links, formatting and minor edits

### PR DESCRIPTION
**Pull request by ericsnekbytes**
Updates the Host Guide and fixes some items from this previous PR (which added the document):

- https://github.com/jupyterlab/team-compass/pull/225

### CODE REVIEW

**Comment by krassowski** [comment]
It looks like there is a conflict. It might be worth re-basing on `main` branch.

**Review by fcollonval — COMMENTED** [review]
Thanks a lot @ericsnekbytes 

I also see that the file `host_guide.rst` is not listed in the `index.rst` and therefore is not accessible on the documentation website. Would you be able to fix that too here please?

**Comment by ericsnekbytes** [comment]
> Thanks a lot @ericsnekbytes
> 
> I also see that the file `host_guide.rst` is not listed in the `index.rst` and therefore is not accessible on the documentation website. Would you be able to fix that too here please?

I will add that! I made other formatting and navigation/wayfinding improvements (and added a thank you passage) as well.

**Comment by ericsnekbytes** [comment]
@krassowski I merged and fixed the conflicts. Do you know if the index.rst addition will work here? It's linking to this host_guide document which I converted to markdown...

**Comment by ericsnekbytes** [comment]
@krassowski @fcollonval The link on the main page should be good, I just checked the latest build. Thanks for the reviews :)

File: docs/index.rst [hunk 1]
```diff
@@ -22,6 +22,7 @@ governance and planning.
    team/becoming-member
    team/decision-making
    team/member-guide
+   host_guide
```
Comment (krassowski) [inline docs/index.rst:25]: Should we switch `host-guide` to harmonise with others?

**Review by krassowski — APPROVED** [review]
Thanks!

**Pull request merged**